### PR TITLE
[Py] Add PartialShape and Dimension to Python API

### PR DESCRIPTION
**Pull request by postrational**
Adding `PartialShape` to the Python API. 

Usage (see tests):

```python
    ps = PartialShape([1, 2, 3, -1])
    assert repr(ps) == '<PartialShape: {1,2,3,?}>'

    ps = PartialShape.dynamic()
    assert repr(ps) == '<PartialShape: ?>'

    ps = PartialShape.dynamic(r=Dimension(2))
    assert repr(ps) == '<PartialShape: {?,?}>'

    ps = PartialShape([Dimension(1), Dimension(2), Dimension(3), Dimension.dynamic()])
    assert repr(ps) == '<PartialShape: {1,2,3,?}>'

    shape = Shape([1, 2, 3])
    ps = PartialShape(shape)
    assert repr(ps) == '<PartialShape: {1,2,3}>'

    ps = PartialShape([1, 2, 3, 4])
    assert repr(ps) == '<PartialShape: {1,2,3,4}>'
```

### CODE REVIEW

**Comment by postrational** [comment]
https://gitlab-icv.inn.intel.com/inference-engine/dldt/merge_requests/7427

File: python/src/pyngraph/partial_shape.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    shape.def("get_min_shape", &ngraph::PartialShape::get_min_shape);
+    shape.def("get_shape", &ngraph::PartialShape::get_shape);
+    shape.def("to_shape", &ngraph::PartialShape::to_shape);
+
+    shape.def("__eq__",
+              [](const ngraph::PartialShape& a, const ngraph::PartialShape& b) { return a == b; },
+              py::is_operator());
+    shape.def("__eq__",
+              [](const ngraph::PartialShape& a, const ngraph::Shape& b) { return a == b; },
+              py::is_operator());
```
Comment (arogowie-intel) [inline python/src/pyngraph/partial_shape.cpp:69]: These equal operators are untested.

File: python/src/pyngraph/partial_shape.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    shape.def_property_readonly("rank", &ngraph::PartialShape::rank);
+    shape.def_property_readonly("all_non_negative", &ngraph::PartialShape::all_non_negative);
+    shape.def_property_readonly("rank", &ngraph::PartialShape::rank);
+
+    shape.def("compatible", &ngraph::PartialShape::compatible);
+    shape.def("refines", &ngraph::PartialShape::refines);
+    shape.def("relaxes", &ngraph::PartialShape::relaxes);
+    shape.def("same_scheme", &ngraph::PartialShape::same_scheme);
+    shape.def("get_max_shape", &ngraph::PartialShape::get_max_shape);
+    shape.def("get_min_shape", &ngraph::PartialShape::get_min_shape);
```
Comment (arogowie-intel) [inline python/src/pyngraph/partial_shape.cpp:60]: This is untested.

File: python/src/pyngraph/partial_shape.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    shape.def(py::init<const std::initializer_list<ngraph::Dimension>&>());
+    shape.def(py::init<const std::vector<ngraph::Dimension>&>());
+    shape.def(py::init<const ngraph::Shape&>());
+
+    shape.def_static("dynamic", &ngraph::PartialShape::dynamic, py::arg("r") = ngraph::Dimension());
+
+    shape.def_property_readonly("is_dynamic", &ngraph::PartialShape::is_dynamic);
+    shape.def_property_readonly("is_static", &ngraph::PartialShape::is_static);
+    shape.def_property_readonly("rank", &ngraph::PartialShape::rank);
+    shape.def_property_readonly("all_non_negative", &ngraph::PartialShape::all_non_negative);
```
Comment (arogowie-intel) [inline python/src/pyngraph/partial_shape.cpp:52]: This function is untested.

File: python/src/pyngraph/dimension.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    dim.def_property_readonly("is_dynamic", &ngraph::Dimension::is_dynamic);
+    dim.def_property_readonly("is_static", &ngraph::Dimension::is_static);
+
+    dim.def("__eq__",
+            [](const ngraph::Dimension& a, const ngraph::Dimension& b) { return a == b; },
+            py::is_operator());
+    dim.def("__eq__",
+            [](const ngraph::Dimension& a, const int64_t& b) { return a == b; },
+            py::is_operator());
```
Comment (arogowie-intel) [inline python/src/pyngraph/dimension.cpp:49]: These operators are untested.

File: python/test/ngraph/test_core.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    assert not d1.refines(d2)
+    assert d1.relaxes(d2)
+    assert d2.refines(d1)
+    assert not d2.relaxes(d1)
+    assert d2.compatible(d1)
+    assert not d2.same_scheme(d1)
+
+    d1 = Dimension(3)
+    d2 = Dimension(3)
+    assert d1.refines(d2)
```
Comment (tomdol) [inline python/test/ngraph/test_core.py:67]: The `refines` and `relaxes` checks are usually done for a mix of static and dynamic dimensions. It's ok to leave them here, but from the values point of view `3` doesn't really refine or relax another `3`.

**Pull request merged**